src/app/new-page/component/tract-page/tract-page.component.ts

```diff
@@ -0,0 +1,285 @@
+      // Set type as Any so we can edit isHidden property.
+      const card_to_show = this.cards[this._cardShownOnFormAction];
+      (card_to_show as any).isHidden = false;
+      this.pageService.formVisible();
+      this.pageService.modalHidden();
+
+      if (card_to_show.label?.text) {
+        this.pageService.changeHeader(card_to_show.label.text);
+      }
+
+      this.cards.forEach((card) => {
+        if (card.listeners?.length) {
+          card.listeners.forEach((listener) => {
+            if (!card.isHidden && listener.name !== functionName) {
+              this._cardsHiddenOnFormAction.push(card.position);
+            }
+          });
+        } else {
+          this._cardsHiddenOnFormAction.push(card.position);
+        }
+      });
+
+      if (this._cardsHiddenOnFormAction.length) {
+        this.cards
+          .filter((card) =>
+            this._cardsHiddenOnFormAction.includes(card.position)
+          )
+          .map((card) => {
+            (card as any).isHidden = true;
+          });
+      }
+    } else if (isHideCard) {
+      this.next();
+      setTimeout(() => {
+        this.pageService.formHidden();
+        this.pageService.modalHidden();
+        if (this._cardsHiddenOnFormAction.length) this.setHiddenCardToShow();
+        if (this._cardShownOnFormAction >= 0) this.setShownCardToHidden();
+
+        this._cardShownOnFormAction = -1;
+        this._cardsHiddenOnFormAction = [];
+      }, 0);
+    } else if (isShowModal) {
+      this.pageService.modalVisible();
+      this.pageService.formHidden();
+    } else if (isHideModal) {
+      this.pageService.modalHidden();
+      this.pageService.formHidden();
+      this.next();
+      setTimeout(() => {
+        if (this._cardsHiddenOnFormAction.length) this.setHiddenCardToShow();
+        if (this._cardShownOnFormAction >= 0) this.setShownCardToHidden();
+
+        this._cardShownOnFormAction = -1;
+        this._cardsHiddenOnFormAction = [];
+      }, 0);
+    } else {
+      this.pageService.contentEvent(functionName);
+    }
+  }
+
+  private setHiddenCardToShow(): void {
+    this.cards
+      .filter((card) => this._cardsHiddenOnFormAction.includes(card.position))
+      .map((card) => {
+        (card as any).isHidden = false;
+      });
+  }
+
+  private setShownCardToHidden(): void {
+    (
+      this.cards.find(
+        (card) => card.position === this._cardShownOnFormAction
+      ) as any
+    ).isHidden = true;
+  }
+
+  private init(): void {
+    this.pageService.setPageOrder(this.order, this.totalPages);
+    this.pageService.modalHidden();
+    this.pageService.formHidden();
+    this.header = this._page.header || null;
+    this.hasPageHeader = !!this._page.header?.title?.text;
+    this.hero = this._page.hero || null;
+    this.cards = this._page.cards || [];
+    this.modal = this._page.modals ? this._page.modals[0] : null;
+    this.callToAction = !!(
+      this.page.callToAction?.label?.text || this.page.callToAction?.tip
```

Comment:
you probably don't need the check for a `tip` since we aren't supporting any tip logic on the web